src/expertsystem/reaction/topology.py

```diff
@@ -61,7 +61,7 @@ def __init__(
         if nodes is not None:
             self.__nodes = set(nodes)
         if edges is not None:
-            self.__edges = edges
+            self.__edges = dict(edges)
```

Comment:
May want to change the type hint into
```python
edges: Optional[Mapping[int, Edge]] = None
```
because the cast into `dict` allows you to give more general mapping structures as input (e.g. using `zip`).

Reply:
done